src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -89,19 +94,45 @@ export async function deleteDataSourceById(
   return savedObjectsClient.delete('data-source', id);
 }
 
+export async function deleteMultipleDataSources(
+  savedObjectsClient: SavedObjectsClientContract,
+  selectedDataSources: DataSourceTableItem[]
+) {
+  await Promise.all(
+    selectedDataSources.map(async (selectedDataSource) => {
+      await deleteDataSourceById(selectedDataSource.id, savedObjectsClient);
+    })
+  );
+}
+
+export async function createNewCredential(
+  savedObjectsClient: SavedObjectsClientContract,
+  newCredential: CreateNewCredentialType
+) {
+  return (
+    savedObjectsClient.create('credential', newCredential).then((response) => response.id || '') ||
+    null
```

Comment:
what's the use of `|| null` here for a server side call?

Reply:
if there is no response from the server, shouldn't it be an error?

Reply:
I mean the response may not contain the expected object. It could just give generic success response.